autoload/wiki/complete.vim

```diff
@@ -99,10 +100,13 @@ function! s:completer_wikilink.complete_anchor(regex) dict abort " {{{2
   let l:length = strlen(l:base)
 
   let l:anchors = wiki#toc#gather_anchors(l:url)
-  call filter(l:anchors, 'v:val =~# ''^'' . wiki#u#escape(l:base) . ''[^#]*$''')
+  call filter(l:anchors, {_,x -> match(x,
+        \ (g:wiki_completion_case_sensitive ? '\C' : '\c')
+        \ . '^' . wiki#u#escape(l:base) . '[^#]*$') >= 0})
   call map(l:anchors, 'strpart(v:val, l:length)')
   if !empty(a:regex)
-    call filter(l:anchors, 'v:val =~# ''' . a:regex . '''')
+    call filter(l:anchors, {_,x -> match(x,
+        \ (g:wiki_completion_case_sensitive ? '\C' : '\c') . a:regex) >= 0})
```

Comment:
I think here and at line 103 it would be better to swap `=~#` with `=~?` for the case sensitive stuff. E.g.:

```vim
     call filter(l:anchors, printf('v:val =~%s ''%s''',
           \ g:wiki_completion_case_sensitive ? '#' : '?',
           \ a:regex))
```

Reply:
Any reason to prefer a string over a lambda? I tend to like the lambas better for readability - the excessiv number of single-quotes necessary bothers me.

Reply:
No, not really; if it is a more or less trivial string I often prefer it, but when it becomes more complex, I agree that lambdas are usually much clearer.